### PR TITLE
Project Movies by Johan Löfgren and Linnéa Helén

### PR DESCRIPTION
**Pull request by Polkapetunian**
Link to my deployed project: https://top-rated-movies-sweden.netlify.app/

My reflections on how this week's project turned out: It turned out good and meeting blue requirements.

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+        </Route>
+        <Route path="/movies/:id">
+          <MovieDetails/>
+        </Route>
+      </Switch>
+    </BrowserRouter>
+  );
+};
+
+//pages here within routes the components are within the pages
```
Comment (Marielle8) [inline code/src/App.js:23]: 😵?

File: code/src/components/LoadingSpinner.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+export const LoadingSpinner = () => {
```
Comment (Marielle8) [inline code/src/components/LoadingSpinner.js:3]: Maybe rename it to Loader so you don't expect some spinning to happen.

File: code/src/components/MovieCard.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      <Link to={`/movies/${id}`} className="movie-poster-wrapper">       
+        <img className="movie-poster" src={`https://image.tmdb.org/t/p/w500/${poster_path}`} alt={title}/>
+        <div className="movie-info">
+          <h2 className="movie-title">{title}</h2>
+          <p className="movie-released">Released: {release_date}</p>
+        </div> 
+      </Link>     
+      
+  )
+}
```
Comment (Marielle8) [inline code/src/components/MovieCard.js:15]: You should have an empty line in the end of each document.

File: code/src/index.css [hunk 4]
```diff
@@ -11,3 +16,131 @@ code {
...
+
+.movie-info:hover {
+  opacity: 1;
+}
+
+.detail-container {
+  position: relative;
+}
+
+.detail-backdrop-container::after {
```
Comment (Marielle8) [inline code/src/index.css:71]: Nice use of :after!

File: code/src/index.css [hunk 5]
```diff
@@ -11,3 +16,131 @@ code {
...
+  margin:0 0 -10px 0;
+  background-image: linear-gradient(to bottom, rgba(255, 255, 255, 0)0, rgb(0, 0, 0) 100%);
+}
+
+.detail-backdrop {
+width: 100vw;
+height: 100vh;
+object-fit: cover;
+margin-bottom: -10px;
+}
```
Comment (Marielle8) [inline code/src/index.css:87]: Use indentation.

File: code/src/pages/MovieDetails.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  const [details, setDetails] = useState();
+
+  useEffect(() => {
+    fetch(DETAILS_URL(id))
+    .then((response) => response.json())
+    .then((json) => setDetails(json));
+  }, []);
+
+  if (details === undefined) {
+    return <LoadingSpinner/>;
```
Comment (Marielle8) [inline code/src/pages/MovieDetails.js:19]: Good that you are showing some kind of loading page here!

File: code/src/pages/MovieDetails.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  return (
+    <div className="detail-container">
+      <Link to={"/"}>
+        <div className="back-link"><img src= "/assets/backarrow50px.png" className="back-link-icon" alt="Go back to list of movies" /><h2 className="back-link-text">Movies</h2></div>
```
Comment (Marielle8) [inline code/src/pages/MovieDetails.js:25]: Fix indentation

File: code/src/pages/MovieDetails.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      </Link>
+      <div className="detail-backdrop-container">
+        <img className="detail-backdrop" src={`https://image.tmdb.org/t/p/original/${details.backdrop_path}`} alt={details.title}/>
```
Comment (Marielle8) [inline code/src/pages/MovieDetails.js:28]: You could save this url to a variable like you are doing with the API call.

File: code/src/pages/MovieList.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  useEffect(() => {
+    fetch(TOP_RATED_URL)
+    .then(response => response.json()
+    .then(json => setMovies(json.results)))
+  }, [])
+  return (
+    <div className="movie-list">
+      {movies.map(movie => {
+        return (
+          <MovieCard key={movie.id} {...movie}/>
```
Comment (Marielle8) [inline code/src/pages/MovieList.js:16]: Spread operator 😎!

File: code/src/pages/MovieList.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React, {useState, useEffect} from 'react'
+import { MovieCard } from '../components/MovieCard'
+import { TOP_RATED_URL } from '../reusable/urls'
+export const MovieList = () => {
+  
+  const [movies, setMovies] = useState([]);
```
Comment (Marielle8) [inline code/src/pages/MovieList.js:6]: There should be an empty line between 1 and 2 and between 3 and 4, useState should be on the first line inside the component.

File: code/src/reusable/urls.js [hunk 11]
```diff
@@ -0,0 +1,3 @@
+export const TOP_RATED_URL = "https://api.themoviedb.org/3/movie/top_rated?api_key=e251592f16990230853a7c79c14797f4&region=SE"
+export const DETAILS_URL = (id) => `https://api.themoviedb.org/3/movie/${id}?api_key=e251592f16990230853a7c79c14797f4&language=en-US` 
+//https://api.themoviedb.org/3/movie/{movie_id}?api_key={api_key}&language=en-US
```
Comment (Marielle8) [inline code/src/reusable/urls.js:3]: Remove line 3.

**Review by Marielle8 — APPROVED** [review]
- Blue goals reached 👍
- Very descriptive classnames! 🎉 
- Not sure if it matters but line spacing could be cleaned up to only one line space between certain code blocks and one line space at the end of a comonent file 🤓
- Nice idea that you took the top-rated movies in Sweden 👍

File: code/src/App.js (hunk 1 above)
Comment (KarolCislowski) [reply code/src/App.js:23]: image

**Review by KarolCislowski — APPROVED** [review]
Great Project 🦄

**Pull request closed without merging**